### ISSUE TITLE
Missing return in periodicFetchData ?

### ISSUE THREAD

**Issue by chrissearle** [airgradienthq/arduino#9]
First up - the code compiles fine in the usual arduino IDE.

But - in platformio - where it builds with certain warnings enabled - in this case -Werror=return-type - we get:

```
AirGradient.cpp: In member function 'TMP_RH AirGradient::periodicFetchData()':
AirGradient.cpp:306:16: error: control reaches end of non-void function [-Werror=return-type]
  306 |     returnError(error);
```

I can fix it locally by just changing line 306 to:

```
return returnError(error);
```

I guess that the arduino IDE doesn't enable that warning.

But - it does look like a return on that line would be a good idea?

Not entirely sure - but thought I'd mention it.

**Comment by Nightreaver** [comment]
I can confirm this issue - it should be fixed for platformio if it doesnt bother arduino IDE anyways

**Comment by airgradienthq** [comment]
Thank you. It has been merged and is part of the latest release (1.4.0)
